src/Microsoft.Azure.Mobile.Client/Platforms/uwp/Authentication/AuthenticationHelper.cs

```
@@ -0,0 +1,72 @@
+
+        /// <summary>
+        /// Occurs when there an error is encountered when authenticating.
+        /// </summary>
+        public event EventHandler<AuthenticatorErrorEventArgs> Error;
+
+        internal void OnResponseReceived(Uri uri)
+        {
+            if (!uri.Fragment.Contains("authorization_code"))
+            {
+                Error?.Invoke(this, new AuthenticatorErrorEventArgs("Authorization code not found in server response"));
+            }
+            else
+            {
+                var fragments = decodeResponse(uri.Fragment);
+                Completed?.Invoke(this, new AuthenticatorCompletedEventArgs(fragments["authorization_code"]));
+            }
+        }
+
+        private Dictionary<string, string> decodeResponse(string encodedString)
+        {
+            var inputs = new Dictionary<string, string>();
+
+            if (encodedString.Length > 0)
+            {
+                char firstChar = encodedString[0];
+                if (firstChar == '?' || firstChar == '#')
+                {
+                    encodedString = encodedString.Substring(1);
+                }
+
+                if (encodedString.Length > 0)
+                {
+                    var parts = encodedString.Split(new char[] { '&' }, StringSplitOptions.RemoveEmptyEntries);
+
+                    foreach (var part in parts)
+                    {
+                        var equalsIndex = part.IndexOf('=');
+
+                        string key;
+                        string value;
+                        if (equalsIndex >= 0)
+                        {
+                            key = Uri.UnescapeDataString(part.Substring(0, equalsIndex));
+                            value = Uri.UnescapeDataString(part.Substring(equalsIndex + 1));
+                        }
+                        else
+                        {
+                            key = Uri.UnescapeDataString(part);
```

I don't think query parameters allow *empty* values, so we should either ignore paramter or throw exception instead of assigning empty string as value. I prefer throwing exception in this case.